src/main/scala/com.ariskk.flink4s/WindowedStream.scala

```diff
@@ -16,4 +18,21 @@ final case class WindowedStream[T, K, W <: Window](stream: JavaWStream[T, K, W])
     DataStream(stream.reduce(reducer))
   }
 
+  def aggregate[A, O](aggregateF: (A, T) => A)(outputF: A => O)(implicit monoid: Monoid[A],
+                                                                aggTypeInformation: TypeInformation[A],
+                                                                typeInformation: TypeInformation[O]): DataStream[O] = {
+    val reducer = new AggregateFunction[T, A, O] {
+
+      override def createAccumulator(): A = Monoid.empty[A]
```

Comment:
Though this sounds like a good idea, I have never used a Monoid in practice because `Monoid.empty[A]` lacks the context of a key (that's needed to do anything meaningly aggregation at scale).
If you find it useful though, happy to have it.

Reply:
Gonna remove taht in the end, I think it makes less sense here.